### ISSUE TITLE
Feature request: tap_action

### ISSUE THREAD

**Issue by Eckart0** [AlexxIT/WebRTC#251]
Hi.
Thank you first of all for this quick camera connection.
For the integration in Lovelace I have the following request: Is it possible to provide the item "tap_action" - as with "picture-entity" - so that one can then navigate or save the current picture? E.G.:
    tap_action:
      action: navigate
      navigation_path: /yaml-dashboard-2/cam_eg
Greetings,
Eckart

**Comment by jzielke84** [comment]
Yes please, this is something I'm missing since it disables basic HassIO ui functionality.

**Comment by sigalou** [comment]
![image](https://user-images.githubusercontent.com/26350534/234974571-f9f75a81-7f57-4bdb-9aca-47b538ff6a9c.png)
https://github.com/junalmeida/homeassistant-webrtc-card

I see it on this version, it is not integrated here? two versions in parallel? THANKS

**Comment by nprez83** [comment]
I'd love to have tap and hold actions supported, if not for the card itself, perhaps for the shortcuts. As of now, I have a volume shortcut that brings up an iframe pointing to the webui using the browser_mod.pop service, but browser_mod service calls open the popup on every registered browser. If I had the ability to use actions for the shortcut, I'd be able to use the fire-dom-event option, which only opens the popup in the current browser.

**Comment by catchdave** [comment]
+1

Use case is showing multiple cameras on screen and wanting a single click to maximize the the feed to fullscreen. Using the built in fullscreen icon has two issues:
1. Requires two taps (one to bring up icons, then to click)
2. Doesn't work with different resolution RTSP streams. My tablet can't handle multiple 4k rtc streams, but can handle a single one, or multiple low res streams. When they are small, there's no need for high res anyway. 

Ideally, I would have multiple low res rtsp cards on screen that when tapped would create a fullscreen single stream of the full resolution RTSP stream (which has a different url).

**Comment by AlexxIT** [comment]
Check this card. It has a lot of functionality
https://github.com/dermotduffy/frigate-hass-card

**Comment by catchdave** [comment]
Wow - that card is incredible and has a lot of functionality. It does seem overkill for what I want, but perhaps that is the only option until this feature request gets prioritized...

Either way, thanks for the fast reply!

**Comment by hapklaar** [comment]
> Check this card. It has a lot of functionality https://github.com/dermotduffy/frigate-hass-card

That's actually a very good card, but unfortunately it uses javascript to go to full screen, which is not supported on iOS. The full screen button in the WebRTC implementation does work, but cannot be triggered in the frigate card.

**Comment by embed-me** [comment]
+1

**Comment by bcutter** [comment]
The only way to use `confirmation` (https://www.home-assistant.io/dashboards/actions/#options-for-confirmation) with text is to use native `tap_action` (and `double_tap_action` as well as `hold_action`).

Therefore please implement the `tap_action`. I want to show a confirmation like "Do you want to start the 30 seconds recording?" before actually running the `service: camera.record`.

**Comment by codemunkie15** [comment]
Is there any movement on this please? Would love to use browser_mod to popup a high res full-screen stream when I click the card

**Comment by AlexxIT** [comment]
Looks like no one wants to implement this.

**Comment by codemunkie15** [comment]
> Looks like no one wants to implement this.

If anyone can point me in the right direction I would be happy to give it a go?

**Comment by codemunkie15** [comment]
I've implemented `tap_action` support in the below PR if you can take a look please.

https://github.com/AlexxIT/WebRTC/pull/668

**Comment by michaelarnauts** [comment]
I wanted this feature just to show a larger video by clicking the whole card, like it works on a picture entity card of Home Assistant.

Something similar could be done by playing with the UI controls and some custom styling:

```yaml
type: custom:webrtc-camera
url: oprit
poster: camera.oprit
ui: true
muted: true
style: |
  .mode { display: none; }
  .screenshot { display: none; }
  .space { display: none; }
  .pictureinpicture { display: none; }
  .controls { top: 0; left: 0; right: 0; bottom: 0; opacity: 0; }
  .fullscreen { width: 100%; }
  ha-card { border: none !important; border-radius: 0px !important; }
```

It configures the UI to be visible but then hides most of the controls except for the fullscreen one. This will be full with and transparent, so you don't see the icon.

The ha-card line is a fix for the white border that is visible in fullscreen mode. I got it from https://github.com/AlexxIT/WebRTC/issues/596#issuecomment-2294932685

**Comment by AndyMillne** [comment]
+1 for this request please. My use case would be to open the camera native app (Unifi Protect).

**Comment by member1980** [comment]
+1 from her - for Browser_Mod-PopUp :)

**Comment by member1980** [comment]
> I wanted this feature just to show a larger video by clicking the whole card, like it works on a picture entity card of Home Assistant.
> 
> Something similar could be done by playing with the UI controls and some custom styling:
> 
> type: custom:webrtc-camera
> url: oprit
> poster: camera.oprit
> ui: true
> muted: true
> style: |
>   .mode { display: none; }
>   .screenshot { display: none; }
>   .space { display: none; }
>   .pictureinpicture { display: none; }
>   .controls { top: 0; left: 0; right: 0; bottom: 0; opacity: 0; }
>   .fullscreen { width: 100%; }
>   ha-card { border: none !important; border-radius: 0px !important; }
> It configures the UI to be visible but then hides most of the controls except for the fullscreen one. This will be full with and transparent, so you don't see the icon.
> 
> The ha-card line is a fix for the white border that is visible in fullscreen mode. I got it from [#596 (comment)](https://github.com/AlexxIT/WebRTC/issues/596#issuecomment-2294932685)

thanks, buddy!

**Comment by IvolveBiz** [comment]
+1 please add
